api/v1beta2/bucket_types.go

```diff
@@ -124,6 +126,11 @@ type BucketStatus struct {
 	// +optional
 	Artifact *Artifact `json:"artifact,omitempty"`
 
+	// IgnoreChecksum is a SHA256 checksum of the excluded patterns applied
+	// in the last reconciliation.
```

Comment:
Maybe
```suggestion
	// IgnoreChecksum is a checksum of .spec.ignore in .status.observedGeneration
	// version of the object.
```
to keep the idea of reconciliation and hashing algorithm out of the API, as they are implementation details.

This suggestion would apply to GitRepo API as well.

Reply:
Also, maybe an optional thing, since this will be the first hash in the status, let's also define the format.
In a previous discussion with Hidde, we thought it'd be good to have the hashing algorithm as part of the value, like `sha256:<checksum>`. The reader can then read the first part and then know how to read the checksum, avoiding any cryptic errors when the checksum decoding fails.

With that, the field description can also have something like:
```
// It is formatted as `<algo>:<checksum>`. For example: `sha256:<checksum>`.
```

Reply:
That's a great idea! Please, check if it is now as you thought.